File: 0605-can-place-flowers/0605-can-place-flowers.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        int count=0;
        int c=0;
        if(flowerbed.size()==1 && flowerbed[0]==0)
            return true;
        
        for(int i=0; i<flowerbed.size(); i++){
            if(flowerbed[i]==0){
                c++;
               if(i==1 || i==flowerbed.size()-1){
                   if(c==2){
                        count++;
                  c=0;
                  i--;
                   }
               }
              if(c==3)
              {
                  count++;
                  c=0;
                  i--;
                  
              }
            }
            else{
                c=0;
            }
        }
        if(count>=n)
            return true;
        return false;
    }
};

File: 0605-can-place-flowers/0605-can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        if len(flowerbed)==1 and flowerbed[0]==0:
                   return True
        count=0
        c=0
        i=0
        while i<len(flowerbed):
                if flowerbed[i]==0:
                    c+=1
                    if i==1 or i==len(flowerbed)-1:
                          if c==2:
                                count+=1
                                c=0
                                i-=1
                    if c==3:
                        count+=1
                        c=0
                        i-=1
                else:
                    c=0
                i+=1
        if count>=n:
             return True
        return False